pandas/core/frame.py

```diff
@@ -472,8 +472,12 @@ class DataFrame(NDFrame, OpsMixin):
         Index to use for resulting frame. Will default to RangeIndex if
         no indexing information part of input data and no index provided.
     columns : Index or array-like
-        Column labels to use for resulting frame. Will default to
-        RangeIndex (0, 1, 2, ..., n) if no column labels are provided.
+        Columns to select from data or column labels to use for resulting frame.
+        Will use the provided labels for the column index if data does not
+        include column labels, defaulting to RangeIndex(0, 1, 2, ..., n).
```

Comment:
I think we can be a little more concise; what do you think of:

Column labels to use for resulting frame when data does not have them, defaulting to RangeIndex(0, 1, 2, ..., n). If data contains column labels, will perform column selection instead.

Reply:
Yes, thats better! Adjusted